File: Problem solving with c++/Problem solving Level 2/Problem10.cpp
/*
Problem10 |===============================================

Write a program to read a number and print it in order from 
left to right.

Input : 
1234

Output : 
1
2
3
4
==========================================================
*/

#include <iostream>
#include <string>
using namespace std;

// Function to read a positive number from the user
int ReadPositiveNumber(string Message)
{
    int Number = 0;
    do
    {
        cout << Message << endl; // Display message prompt
        cin >> Number;           // Read user input
    } while (Number <= 0);       // Ensure input is positive
    return Number;
}

// Function to reverse the digits of a given number
int ReverseNumber(int Number)
{
    int Remainder = 0, Number2 = 0;
    while (Number > 0)
    {
        Remainder = Number % 10;       // Extract last digit
        Number = Number / 10;          // Remove last digit from number
        Number2 = Number2 * 10 + Remainder; // Append digit to reversed number
    }
    return Number2;
}

// Function to print the digits of a number one by one
void PrintDigits(int Number)
{
    int Remainder = 0;
    while (Number > 0)
    {
        Remainder = Number % 10; // Extract last digit
        Number = Number / 10;    // Remove last digit
        cout << Remainder << endl; // Print the digit
    }
}

int main()
{
    // Read a positive number, reverse it, and print its digits
    PrintDigits(ReverseNumber(ReadPositiveNumber("Please enter a positive number ? ")));
    return 0;
}

File: Problem solving with Python/Problem10.py
"""
Problem10 |===============================================

Write a program to read a number and print it in order from 
left to right.

Input : 
1234

Output : 
1
2
3
4
==========================================================
"""

def read_positive_number(message):
    """
    Reads a positive integer from the user.
    """
    number = 0
    while number <= 0:
        try:
            number = int(input(message + "\n"))
        except ValueError:
            print("Invalid input. Please enter a positive integer.")
    return number


def reverse_number(number):
    """
    Reverses the digits of the given number.
    """
    reversed_number = 0
    while number > 0:
        remainder = number % 10  # Extract last digit
        number = number // 10  # Remove last digit
        reversed_number = reversed_number * 10 + remainder  # Append digit to reversed number
    return reversed_number


def print_digits(number):
    """
    Prints the digits of a number one by one.
    """
    while number > 0:
        remainder = number % 10  # Extract last digit
        number = number // 10  # Remove last digit
        print(remainder)


if __name__ == "__main__":
    # Read a positive number, reverse it, and print its digits
    print_digits(reverse_number(read_positive_number("Please enter a positive number: ")))